core/schema/service.go

```diff
@@ -105,14 +123,87 @@ func (s *Service) Create(ctx context.Context, nsName string, schemaName string,
 		Compatibility: compatibility,
 	}
 	versionID := getIDforSchema(nsName, schemaName, sf.ID)
+	_, prevSchemaData, _ := s.GetLatest(ctx, nsName, schemaName)
 	version, err := s.repo.Create(ctx, nsName, schemaName, mergedMetadata, versionID, sf)
+	changeRequest := &changedetector.ChangeRequest{
+		NamespaceID: nsName,
+		SchemaName:  schemaName,
+		Version:     version,
+		VersionID:   versionID,
+		OldData:     prevSchemaData,
+		NewData:     data,
+		Depth:       s.config.SchemaChange.Depth,
+	}
+	newCtx := context.Background()
+	go s.identifySchemaChange(newCtx, changeRequest)
 	return SchemaInfo{
 		Version:  version,
 		ID:       versionID,
 		Location: fmt.Sprintf("/v1beta1/namespaces/%s/schemas/%s/versions/%d", nsName, schemaName, version),
 	}, err
 }
 
+func (s *Service) identifySchemaChange(ctx context.Context, request *changedetector.ChangeRequest) error {
+	endFunc := s.newrelic.StartGenericSegment(ctx, "Identify Schema Change")
+	defer endFunc()
+	schemaID, err := s.repo.GetSchemaID(ctx, request.NamespaceID, request.SchemaName)
+	if err != nil {
+		log.Printf("got error while getting schema ID from DB %s", err.Error())
+		return err
+	}
+	prevEvent, err := s.notificationEventRepo.GetByNameSpaceSchemaVersionAndSuccess(ctx, request.NamespaceID, schemaID, request.VersionID, true)
+	if err != nil && !errors.Is(err, pgx.ErrNoRows) {
+		log.Printf("got error while fetching previous notification event status for for namespace : %s, schema: %s, version: %d, %s", request.NamespaceID, request.SchemaName, request.Version, err.Error())
+		// todo push to grafana
```

Comment:
Any errors over here, will be recorded in line 138. We can remove this todo?

Reply:
done